core/src/labels/label.cpp

```diff
@@ -264,31 +260,40 @@ bool Label::updateState(const glm::mat4& _mvp, const glm::vec2& _screenSize, flo
         enterState(State::out_of_screen, 0.0);
     }
 
+    return true;
+}
+
+bool Label::evalState(const glm::vec2& _screenSize, float _dt) {
+
     bool animate = false;
+    m_prevState = m_state;
 
-    switch (m_currentState) {
+    switch (m_state) {
         case State::visible:
-            if (occludedLastFrame) {
-                m_fade = FadeEffect(false, m_options.hideTransition.ease, m_options.hideTransition.time);
+            if (m_occluded) {
+                m_fade = FadeEffect(false, m_options.hideTransition.ease,
+                                    m_options.hideTransition.time);
                 enterState(State::fading_out, 1.0);
                 animate = true;
             }
             break;
         case State::fading_in:
-            if (occludedLastFrame) {
-                enterState(State::sleep, 0.0);
+            if (m_occluded) {
+                // enterState(State::sleep, 0.0);
+                enterState(State::fading_out, m_transform.state.alpha);
+                animate = true;
                 break;
             }
             setAlpha(m_fade.update(_dt));
             animate = true;
-            if (m_fade.isFinished()) {
+            if (m_fade.isFinished() || (m_transform.state.alpha > 0.9f)) {
```

Comment:
Did you notice any transition not working properly while only testing `m_fade.isFinished()`?